src/test/java/org/assertj/core/api/JUnitJupiterBDDSoftAssertionsFailureTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2018 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import org.assertj.core.util.Lists;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+import org.opentest4j.MultipleFailuresError;
+
+import java.util.List;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+@RunWith(JUnitPlatform.class)
+public class JUnitJupiterBDDSoftAssertionsFailureTest {
+
+  // we cannot register the extension here, because we need to test the failure without this test failing!
+  // @RegisterExtension
+  JUnitJupiterBDDSoftAssertions softly = new JUnitJupiterBDDSoftAssertions();
+
+  @Test
+  public void should_report_all_errors() throws Throwable {
```

Comment:
keen to have a test with GIVEN WHEN THEN sections and `catchThrowable` as it makes it clearer what we are testing (the code you have rightfully duplicated was written pre java 8), somthing like:

```java
   // given some preconditions

   // when
   Throwable thrown = catchThrowable(() -> { throw new Exception("boom!"); });

   // then
   assertThat(thrown).isInstanceOf(Exception.class)
                     .hasMessageContaining("boom");
```